### PR TITLE
RavenDB-15918  Support RAVENDB_DATABASE in docker setup 

### PR DESCRIPTION
**Pull request by poissoncorp**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-15918

### Additional description

The changes from this PR let users to provide a default database to be created on startup. If user is storing `RAVEN_DATABASE ` environmental variable, there'll be database created on startup, named after this env var. 

Script-result

It's working with both unsecured and secured servers. Also, I've created a utility script, `get-server-var.sh`, which lets to fetch server variables via rvn admin-channel.

### Type of change

- New feature

### How risky is the change?
- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- Docker

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: docker/ravendb-ubuntu/create-database.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    function usage() {
+        echo "Putdb creates database on Raven.Server via HTTP(S) request, that is named after RAVEN_DATABASE environmental variable."
+        echo "It should take no arguments when working with unsecured server."
+        echo "In the case of secured server it should take path to .crt certificate file as first arg and path to .key file as second to work properly."
+        echo "It returns 1 if cURL fails and 2 if there's wrong input."
+    }
+    [[ $# -eq 0 ]] && mode=unsecured || mode=secured
+    if [[ $mode == unsecured ]]; then
+        if curl -s $serverUrl'/admin/databases?name='$RAVEN_DATABASE -X PUT --compressed --data-raw '{"DatabaseName":"'$RAVEN_DATABASE'"}' > /dev/null; then 
+        echo "Database $RAVEN_DATABASE created successfully on unsecured server."
```
Comment (ayende) [inline docker/ravendb-ubuntu/create-database.sh]:
```suggestion
        echo "Database '$RAVEN_DATABASE' created successfully on unsecured server."
```

File: docker/ravendb-ubuntu/create-database.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        else
+        echo "Database $RAVEN_DATABASE wasn't created successfully - see error output above for details."
```
Comment (ayende) [inline docker/ravendb-ubuntu/create-database.sh]:
```suggestion
        echo "Database '$RAVEN_DATABASE' wasn't created successfully - see error output above for details."
```

**Review by ayende — COMMENTED** [review]
Minor - make the output slightly nicer

File: docker/ravendb-ubuntu/get-server-url.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+# This script takes scheme as first arg (http, https, tcp etc.)
+
+[[ $# -lt 1 ]] && scheme="http" || scheme=$1
```
Comment (gregolsky) [inline docker/ravendb-ubuntu/get-server-url.sh]: can we put get-server-url and get-server-var stuff in a single `server-utils.sh` file?
One less file to worry about and if we wanted to add something it's just a function away.

File: docker/ravendb-ubuntu/run-raven.sh [hunk 4]
```diff
@@ -10,6 +10,11 @@ if [ ! -z "$RAVEN_ARGS" ]; then
...
-eval $COMMAND
-exit $?
+[ -n "$RAVEN_DATABASE" ] && export RAVEN_Setup_Mode=None
+$COMMAND &
+COMMANDPID=$!
+
+[ -n "$RAVEN_DATABASE" ] && ./create-database.sh
 
+wait $COMMANDPID
+exit $?
```
Comment (gregolsky) [inline docker/ravendb-ubuntu/run-raven.sh]: `exit $?` is redundant. wait pid is going to exit with the exit code of the waited process

File: docker/ravendb-ubuntu/create-database.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        exit 1
+        fi
+    else
+        if [[ $mode == secured ]]; then
+            public_uri=$(./get-server-var.sh Configuration.Core.PublicServerUrl.UriValue)
+            public_tcp_uri=$(./get-server-var.sh Configuration.Core.PublicTcpServerUrl.UriValue)
+            if curl -s "$serverUrl/admin/databases?name=$RAVEN_DATABASE" -X PUT -k \
+                --cert $1 \
+                --key $2 \
+                --resolve $public_uri:127.0.0.1 \
```
Comment (gregolsky) [inline docker/ravendb-ubuntu/create-database.sh]: this 127 here might be problematic sometimes - it is going to only work if we're using `0.0.0.0` or `127.0.0.1` in the container. If somebody explicitly binds to e.g. container's internal address then we won't be able to reach it. I think we should only use `--resolve` when `ServerUrl` is either `0.0.0.0` or `127.0.0.1` - in other cases we should just try using whatever DNS gives us.

File: docker/ravendb-ubuntu/create-database.sh (hunk 1 above)
Comment (poissoncorp) [reply docker/ravendb-ubuntu/create-database.sh]: fixed in latest commit

File: docker/ravendb-ubuntu/create-database.sh (hunk 2 above)
Comment (poissoncorp) [reply docker/ravendb-ubuntu/create-database.sh]: fixed in latest commit

File: docker/ravendb-ubuntu/create-database.sh (hunk 5 above)
Comment (poissoncorp) [reply docker/ravendb-ubuntu/create-database.sh]: applied the fix in latest commit

File: docker/ravendb-ubuntu/get-server-url.sh (hunk 3 above)
Comment (poissoncorp) [reply docker/ravendb-ubuntu/get-server-url.sh]: done

File: docker/ravendb-ubuntu/run-raven.sh (hunk 4 above)
Comment (poissoncorp) [reply docker/ravendb-ubuntu/run-raven.sh]: done

File: docker/ravendb-ubuntu/cert-utils.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+function extract_cert_and_key() {
```
Comment (gregolsky) [inline docker/ravendb-ubuntu/cert-utils.sh]: here's underscores and in other places it's spaces?

Comment (poissoncorp) [reply docker/ravendb-ubuntu/cert-utils.sh]: done

File: docker/ravendb-ubuntu/server-utils.sh [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            putdb-usage
+            exit 2
+            fi
+        fi
+    }
+
+wait-for-server
+cert_path=$(get-server-var Configuration.Security.CertificatePath)
+
+if [[ "$cert_path" == "null" ]]; then
```
Comment (gregolsky) [inline docker/ravendb-ubuntu/server-utils.sh:98]: I think we should check if the database exists first - if one would restart the container the script is going to fail.

File: docker/ravendb-ubuntu/server-utils.sh [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            exit 1
+            fi
+        else
+            if [[ $mode == secured ]]; then
+
+                if [[ $serverUrl == "0.0.0.0" || $serverUrl == "127.0.0.1" ]]; then
+                    public_uri=$(get-server-var Configuration.Core.PublicServerUrl.UriValue)
+                    public_tcp_uri=$(get-server-var Configuration.Core.PublicTcpServerUrl.UriValue)
+                    curl_resolve="--resolve $public_uri:127.0.0.1 --resolve $public_tcp_uri:127.0.0.1"
+                    else
```
Comment (gregolsky) [inline docker/ravendb-ubuntu/server-utils.sh]: please reduce indent

Comment (poissoncorp) [reply docker/ravendb-ubuntu/server-utils.sh]: done

File: docker/ravendb-ubuntu/server-utils.sh (hunk 7 above)
Comment (poissoncorp) [reply docker/ravendb-ubuntu/server-utils.sh:98]: done

**Review by gregolsky — APPROVED** [review]

**Pull request merged**